jre/java/java/util/stream/DoubleStream.java

```diff
@@ -240,5 +289,34 @@ static DoubleStream of(double t) {
 
   DoubleSummaryStatistics summaryStatistics();
 
+  default DoubleStream takeWhile(DoublePredicate predicate) {
+    Spliterator.OfDouble original = spliterator();
+    Spliterator.OfDouble spliterator =
+        new Spliterators.AbstractDoubleSpliterator(original.estimateSize(),
+                original.characteristics() & ~(Spliterator.SIZED | Spliterator.SUBSIZED)) {
+          private boolean take = true;
+          private boolean found;
```

Comment:
Seems like you are setting this always first thing in the method, why is this a field?

Reply:
`found` is being updated inside the lambda, so it must be a field, locals used in a lambda or inner class must be final (or "effectively" so).

Reply:
Thanks I saw the lambda now. (The github review UI is an eyesore to me, still trying to get used to.)